.github/workflows/goreleaser.yml

```diff
@@ -1,16 +1,166 @@
+# latest release), we must:
+#
+# 1. Copy the main formula in our Homebrew tap for the previous latest release to a
+#    versioned formula
+# 2. Build and publish the new release to the main formula
+# 3. Update the Homebrew symlink alias for the latest version
+#
+# In the case of the latter (a patch release for an older version), we only need to:
+#
+# 1. Build and publish the new release to a versioned Homebrew formula
+#
+# This action contains 5 jobs to accommodate both cases:
+#
+# release_type    [always]      - checks tags, determines if we're case 1. or case 2.
+# goreleaser_pre  [if case 1.]  - copies the main formula to a versioned formula
+# goreleaser      [always]      - runs tests, builds + publishes with goreleaser
+# goreleaser_post [if case 1.]  - updates symlink to latest version
+# npm             [always]      - publishes to npm
 jobs:
+  # release_type compares the most recently created tag to the highest versioned tag
+  # available on the repo to determine if this release is for a new latest version or a
+  # patch of an older version.
+  release_type:
+    name: Determine release type
+    runs-on: ubuntu-latest
+    outputs:
+      is_latest_version: ${{ env.is_latest_version }}
+      latest_tag: ${{ env.latest_tag }}
+      second_latest_tag: ${{ env.second_latest_tag }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Fetch most recent tag, latest (highest version) tag, and second latest tag
+        # For latest and second latest tags, we can't use git tag --sort=version:refname
+        # because git doesn't have a concept of pre-release versions and thus mis-sorts
+        # versions like 4.0.0-rc.0 *after* 4.0.0.
+        run: |
+          echo "most_recent_tag=$(git tag --sort=taggerdate | tail -1)" >> $GITHUB_ENV
+          echo "latest_tag=$(git tag | tr - \~ | sort -V | tr \~ - | tail -1)" >> $GITHUB_ENV
+          echo "second_latest_tag=$(git tag | tr - \~ | sort -V | tr \~ - | tail -2 | sed -n 1p)" >> $GITHUB_ENV
+      - name: Install semver
+        run: |
+          wget -O /usr/local/bin/semver https://raw.githubusercontent.com/fsaintjacques/semver-tool/master/src/semver
+          chmod +x /usr/local/bin/semver
+      - name: Compare tags
+        # If the most recent tag is also the latest tag, semver compare returns 0. If it's
+        # a patch of an older version, semver compare will return -1.
+        run: |
+          if [[ $(semver compare ${{ env.latest_tag }} ${{ env.most_recent_tag }}) == 0 ]]
+          then
+            echo "is_latest_version=1" >> $GITHUB_ENV
+          else
+            echo "is_latest_version=0" >> $GITHUB_ENV
+          fi
+      - name: Log variables
+        run: |
+          echo "Most recent (current) tag: ${{ env.most_recent_tag }}"
+          echo "Latest version tag: ${{ env.latest_tag }}"
+          echo "Building current version as new latest?: ${{ env.is_latest_version }}"
```

Comment:
I could probably get rid of all this logging but it doesn't hurt to keep it, at least until we actually test an actual release. 🙂